### PR TITLE
[SELS-TASK] Latest activity in profile page

### PR DESCRIPTION
**Pull request by jstephend-sun**
Asana [Task](https://app.asana.com/0/1201478050789792/1201668705387963/f)

expected output:
the latest activity will now also be shown under the profile of the user

screenshots

image

note: on the screenshot, it says followers of 2 because another account is following the user, its not shown in the logs because that user (sir ej) followed the user on the screenshot before the logs were implemented

### CODE REVIEW

File: frontend/src/components/UserProfile/UserProfile.tsx [hunk 1]
```diff
@@ -94,6 +100,23 @@ export const _UserProfile = ({
...
+      const [doer, action, recipient] = JSON.parse(latest.message);
+
+      return (
+        <Stack>
+          {doer} {action} {recipient}
+        </Stack>
+      );
+    }
+
+    return '';
```
Comment (ejanton-sun) [inline frontend/src/components/UserProfile/UserProfile.tsx]:
```suggestion
    if (!latest) return;
      const [doer, action, recipient] = JSON.parse(latest.message);

      return (
        <Stack>
          {doer} {action} {recipient}
        </Stack>
      );
    
```

Comment (ejanton-sun) [reply frontend/src/components/UserProfile/UserProfile.tsx]: apply proper formatting also

File: frontend/src/components/UserProfile/UserProfile.tsx [hunk 2]
```diff
@@ -143,7 +166,18 @@ export const _UserProfile = ({
           </div>
 
           <Divider />
-          <div className="my-4 text-center">Latest Activity</div>
+          {loadingUserData ? (
```
Comment (ejanton-sun) [inline frontend/src/components/UserProfile/UserProfile.tsx]: Typically we use action verbs in loading states.
`isLoadingUserData` more readable than `loadingUserData`

**Review by ejanton-sun — CHANGES_REQUESTED** [review]

File: frontend/src/components/UserProfile/UserProfile.tsx (hunk 2 above)
Comment (jstephend-sun) [reply frontend/src/components/UserProfile/UserProfile.tsx]: done

File: frontend/src/components/UserProfile/UserProfile.tsx (hunk 1 above)
Comment (jstephend-sun) [reply frontend/src/components/UserProfile/UserProfile.tsx]: done

**Review by ejanton-sun — APPROVED** [review]

**Pull request merged**